libraries/Microsoft.Bot.Builder.Dialogs/WaterfallDialog.cs

```diff
@@ -20,12 +20,35 @@ public class WaterfallDialog : Dialog
         private const string StepIndex = "stepIndex";
         private const string PersistedValues = "values";
 
-        private WaterfallStep[] _steps;
+        private List<WaterfallStep> _steps;
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WaterfallDialog"/> class.
+        /// </summary>
+        /// <param name="dialogId">dialog id</param>
+        /// <param name="steps">optional steps to be defined by caller</param>
         public WaterfallDialog(string dialogId, WaterfallStep[] steps = null)
```

Comment:
its kind of awkward to receive an array and convert it to list. Do people create the array just for this class? if yes, maybe receive a list? This is fine too, just thinking what's best.

Reply:
good point, I changed to IEnumerable<> that covers [], list, set, etc.